# Module-12-Gene-annotation
The main purpose of these exercises is to learn how to run command-line based tools to perform ab initio gene predictions. We will predict the positions of genes in an already assembled genome sequence and visualize the resulting predictions in a Genome Browser. We use the well-characterized genome sequence of the Escherichia coli (E. coli) bacterium, since it will allow us to easily check how well our prediction actually performs by comparing it to existing annotations of the genome sequence. Since the programs we are using are designed to be run on the terminal, as is typically the case for large-scale bioinformatics programs, we will once more utilize the Saga system.
The main learning outcomes of this exercise are:
- Using an external server to run a command-line pipeline starting from the raw sequence to final results
- Installing a gene prediction tool (Glimmer) on the server
- Using Python scripts as part of the processing of the data
- Downloading and uploading data to/from the server
- Visualizing and exploring our results in a Genome Browser system

## 12.1 Running Glimmer to predict genes in E. coli
- Go to [these exercises](exercises/Exercise1.md) and follow the steps

## 12.2 Visualzing gene annotation in UCSC Genome Browser
- Go to [these exercises](exercises/Exercise2.md) and follow the steps

## 12.3 Advanced/optional: Create a Python script that computes GC content of the `ecoli.fa` file
- Go to [these exercises](exercises/Exercise3.md) and follow the steps
